Lib/multiprocessing/pool.py

```diff
@@ -177,13 +178,15 @@ def __init__(self, processes=None, initializer=None, initargs=(),
 
         self._worker_handler = threading.Thread(
             target=Pool._handle_workers,
-            args=(self, )
+            args=(self._cache, self._taskqueue, self._ctx, self.Process,
```

Comment:
Such a mouthful :-/ What do you think about gathering most of this in something like a `PoolState` structure?

Reply:
This existed in the previous code from 10 years ago...
https://github.com/python/cpython/blob/1f0b6586387a3dae59cf28239effe14f1333229a/Lib/multiprocessing/pool.py#L137
 I rather fix the bug and not refactor/rewrite the code (if you mean it only for this function I see no point, if you mean it for everything, it would basically amount to rewriting the whole class.

Reply:
I meant for everything. Fair enough, let's keep it like that then.